solr/core/src/test/org/apache/solr/handler/TestUserManagedReplicationWithAuth.java

```diff
@@ -0,0 +1,267 @@
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.params.CommonParams;
+import org.apache.solr.common.params.ModifiableSolrParams;
+import org.apache.solr.embedded.JettySolrRunner;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+@SuppressSSL
+public class TestUserManagedReplicationWithAuth extends SolrTestCaseJ4 {
+  JettySolrRunner leaderJetty, followerJetty, followerJettyWithAuth;
+  SolrClient leaderClient, followerClient, followerClientWithAuth;
+  ReplicationTestHelper.SolrInstance leader = null, follower = null, followerWithAuth = null;
+
+  private static String user = "solr";
+  private static String pass = "SolrRocks";
+  private static String securityJson =
+      "{\n"
+          + "\"authentication\":{ \n"
+          + "   \"blockUnknown\": true, \n"
+          + "   \"class\":\"solr.BasicAuthPlugin\",\n"
+          + "   \"credentials\":{\"solr\":\"IV0EHq1OnNrj6gvRCwvFwTrZ1+z1oBbnQdiVC3otuq0= Ndd7LKvVBAaZIF0QAVi1ekCfAJXr1GGfLtRUXhgrF8c=\"}, \n"
+          + "   \"realm\":\"My Solr users\", \n"
+          + "   \"forwardCredentials\": false \n"
+          + "},\n"
+          + "\"authorization\":{\n"
+          + "   \"class\":\"solr.RuleBasedAuthorizationPlugin\",\n"
+          + "   \"permissions\":[{\"name\":\"security-edit\",\n"
+          + "      \"role\":\"admin\"}],\n"
+          + "   \"user-role\":{\"solr\":\"admin\"}\n"
+          + "}}";
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+    systemSetPropertySolrDisableUrlAllowList("true");
+    // leader with Basic auth enabled via security.json
+    leader =
+        new ReplicationTestHelper.SolrInstance(
+            createTempDir("solr-instance").toFile(), "leader", null);
+    leader.setUp();
+    // Configuring basic auth for Leader
+    Path solrLeaderHome = Path.of(leader.getHomeDir());
+    Files.write(
+        solrLeaderHome.resolve("security.json"), securityJson.getBytes(StandardCharsets.UTF_8));
+    leaderJetty = ReplicationTestHelper.createAndStartJetty(leader);
+    leaderClient =
+        ReplicationTestHelper.createNewSolrClient(
+            buildUrl(leaderJetty.getLocalPort()), DEFAULT_TEST_CORENAME);
+
+    // follower with no basic auth credentials for leader configured.
+    follower =
+        new ReplicationTestHelper.SolrInstance(
+            createTempDir("solr-instance").toFile(), "follower", leaderJetty.getLocalPort());
+    follower.setUp();
+    followerJetty = createAndStartJetty(follower);
+    followerClient =
+        ReplicationTestHelper.createNewSolrClient(
+            buildUrl(followerJetty.getLocalPort()), DEFAULT_TEST_CORENAME);
+
+    // follower with basic auth credentials for leader configured in solrconfig.xml.
+    followerWithAuth =
+        new ReplicationTestHelper.SolrInstance(
+            createTempDir("solr-instance").toFile(), "follower-auth", leaderJetty.getLocalPort());
```

Comment:
The second arg "follower-auth" is a partial file name to "solrconfig-follower-auth.xml"; are you sure this works?  I didn't notice this approach elsewhere; I've seen a whole file name.

Reply:
Nevermind; it does